### PR TITLE
feat(operators): operator browser page

### PR DESCRIPTION
**Pull request by btroncone**
This includes the first version of UI for the operator section, as well as beginning samples for `combineAll` and `combineLatest`. 

The static operator content is split into individual files, I figured this would be easiest for contributors to consume, or it can be extracted into a document db or similar if we choose to go that route in the future. I'm currently working on combining the content and examples from the official docs with relevant examples from learnrxjs.  

Please let me know if you have any additional thoughts or ideas, thanks!

Demo

### CODE REVIEW

**Comment by ladyleet** [comment]
@btroncone awesome! @kwonoj @benlesh can y'all peep at this when you get a chance?

File: src/operator-docs/combination/combineLatest.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { OperatorDoc } from '../operator.model';
+
+export const combineLatest: OperatorDoc = {
```
Comment (benlesh) [inline src/operator-docs/combination/combineLatest.ts]: Presumably this whole document is a stub and we're going to be generating this data from the TS files directly, right? 

We should be able to link the exact line (and SHA) in Github this doc was generated from so people can "read the source"

**Comment by ladyleet** [comment]
@benlesh let's worry about that as we come to it but just get these operators and the official docs so that people can start having a good reference  let's worry about that as we come to it but just get these operators and the official docs so that people can start having a good reference.  The rest can be incremental.

**Comment by ladyleet** [comment]
@benlesh wanting more your feedback on the content?

File: src/operator-docs/combination/combineLatest.ts (hunk 1 above)
Comment (btroncone) [reply src/operator-docs/combination/combineLatest.ts]: Right, I wasn't sure about the long term solution. This is just a temporary hack so I could start getting data on the page / establish structure.

**Comment by benlesh** [comment]
@ladyleet I only wanted to understand the vision/plan for this.

- How will we acquire the data from the code files? 
- How will we migrate to that data?

etc.

**Comment by benlesh** [comment]
Otherwise, it looks good... I don't think I'd hide the "Open in JSBin" links under a menu though, I feel like those should be readily visible to people so they know they can do that.  Even better would be to have embedded playgrounds, but that might be a tall order.

**Comment by btroncone** [comment]
Thanks, I'll move external links out of the menu. Embedded playgrounds is definitely the ultimate goal, @jayphelps brought this up a few days back as well. This and converting marble diagram images to an interactive RxMarbles (or similar) component are the two biggest features I would love to include. I can work on these after finishing other work or maybe we can pick up a contributor to work on this in meantime. @ladyleet

**Comment by btroncone** [comment]
I added a JSBin embed for examples. We can replace later if another option will work better.

**Comment by kwonoj** [comment]
I'm feeling PR grows too large that nearly not able to review. What's plan's here? Originally it says `This isn't ready to merge` - is this means PR will grow further, or this is just poc and will breakdown into separate PR as small chunks?

**Comment by btroncone** [comment]
I think where this stands now is a good base to build off for the operator UI. Operator content and additional UI tweaks can be included in future (much smaller) PR's.

**Comment by kwonoj** [comment]
@btroncone my Q is about this PR itself. it have more than 130 files in single PR, I'm not sure I can follow all of them.

**Comment by ladyleet** [comment]
Can we just yolo this because it's one of the larger more significant PR is and it's not like anything is going to break?

**Comment by ladyleet** [comment]
oops sorry fat fingers :-)

**Comment by btroncone** [comment]
@kwonoj Sure, you can pretty much ignore everything from `operator-docs`, those are just placeholders so I could start getting a sense of structure for operator content and build the menu.

The rest make up the operator browser page / module. I split up each section of each operator into it's own component (ex. `operator-header`, `operator-examples`, etc) so they would be easier to maintain and extend. Most are just small presentation components, the only component with much going on is the one routable component [`operators.component`](https://github.com/ReactiveX/rxjs-docs/pull/34/files#diff-eea5db4aadeae87030ee377021142d12). 

The other new files included are for the `rx-marbles` web component.

Hope this helps, sorry for the massive amount of files. Since this is one big page I wanted to get it to a working state before submitting, my PRs will be much smaller moving forward.

**Comment by kwonoj** [comment]
I still believe splitting this PR into smaller chunks will be helpful, and do not add more stuff into it. First, at least placeholder (/transformation..) can be separated, and others as well like rxMarble integration. And at least main core component needs some test (..I guess? we need test coverage, right?).

**Comment by btroncone** [comment]
Sure, I'll add some tests for the core component.

**Comment by btroncone** [comment]
@kwonoj It looks like you are missing a `-` in prod build, it should be `ng build --prod`. Noticed in the travis build.

https://github.com/ReactiveX/rxjs-docs/blob/f5bf6a8eeda4ef3bf367c8219ea7827ae20fdf0e/.travis.yml#L28

**Comment by kwonoj** [comment]
@btroncone yeah, good catch.

**Comment by ladyleet** [comment]
@btroncone what did we decide on getting this PR in or breaking it into smaller ones or.... <3 thanks for all your hard work!

**Comment by btroncone** [comment]
@ladyleet I think it makes sense to go in as one PR as it's really just one page, everything here fits together. The file creep was caused by the 70+ operator placeholder files and 3 files (ts, html, scss) for each presentation component within each operator. 

I'm pretty much tapped for the weekend so if you need me to split this up I'll have to do it early next week. Let me know what you think, thanks! 👍

**Comment by ladyleet** [comment]
@btroncone meh, let's just merge it. it's out of date with base - can you fix that? no rush if it needs to wait for next week.

**Comment by btroncone** [comment]
Sure, done. 👍

**Comment by ladyleet** [comment]
@btroncone oops looks like travis build failed for some reason

**Comment by btroncone** [comment]
Sorry, it was linting issue. All checks should be passing now. :)

**Comment by ladyleet** [comment]
omg! merging! woot!

**Pull request merged**